src/theory/evaluator.cpp

```diff
@@ -99,6 +100,177 @@ EvalResult Evaluator::evalInternal(TNode n,
 
       switch (currNodeVal.getKind())
       {
+        case kind::CONST_RATIONAL:
+        {
+          const Rational& r = currNodeVal.getConst<Rational>();
+          Assert(r.getDenominator() == 1);
+          results[currNode] = EvalResult(r.getNumerator());
```

Comment:
I think we should just carry around the Rational instead of Integer.  We likely will want to extend this to handle rational arithmetic.